arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ArrowReader.java

```diff
@@ -252,17 +297,11 @@ public boolean hasNext() {
         }
       } catch (IOException | RuntimeException e) {
         if (currentTask != null && !currentTask.isDataTask()) {
-          throw new RuntimeException(
-              "Error reading file: " + getInputFile(currentTask).location() +
-                  ". Reason: the current task is not a data task, i.e. it cannot read data rows. " +
-                  "Ensure that the tasks passed to the constructor are data tasks. " +
-                  "The file scan tasks are: " + fileTasks,
-              e);
-        } else {
-          throw new RuntimeException(
-              "An error occurred while iterating through the file scan tasks or closing the iterator," +
-                  " see the stacktrace for further information. The file scan tasks are: " + fileTasks, e);
+          LOG.error("Error reading file: {}", getInputFile(currentTask).location(), e);
         }
+        // Wrap and throw a Runtime exception because Iterator::hasNext()
+        // cannot throw a checked exception.
+        throw new RuntimeException(e);
```

Comment:
If the exception is already a `RuntimeException` then this should not wrap it in another because that may change the exception type. You can use `ExceptionUtil.castAndThrow(e, RuntimeException.class);` for this.

Reply:
Agreed. ExceptionUtil::castAndThrow was not public and to use it here, I've made it public. I hope that is ok.